javascript/javascript1/week3/series_duration.js

```diff
@@ -0,0 +1,61 @@
+// DATA
+const someSeries = [
+  {
+    title: "Game of thrones",
+    days: 3,
+    hours: 1,
+    minutes: 0,
+  },
+  {
+    title: "Sopranos",
+    days: 3,
+    hours: 14,
+    minutes: 0,
+  },
+  {
+    title: "The Wire",
+    days: 2,
+    hours: 12,
+    minutes: 0,
+  },
+];
+
+//LOGIC
+  function howMuchTime (seriesDurations) {
+    const lifeDays = 29200; // = 80 years
+    const timeProperties = ["days", "hours", "minutes"];
+    const timeConverter = [1, 24, 1440];
+    
+    const percentageBySeriesDays = {};
+    let totalSeriesPercentage = 0;
+
+for (let i = 0; i<seriesDurations.length; i++) {
+    let seriesInDays = 0;
+    for (let j = 0; j<timeProperties.length; j++) {
+        seriesInDays += (seriesDurations[i][timeProperties[j]])/timeConverter[j];
+    }
+
+    let eachSeriesTitle = seriesDurations[i]["title"];
+    let eachSeriesPercentage = (seriesInDays/lifeDays*100).toFixed(3);
+
+    percentageBySeriesDays[eachSeriesTitle] = eachSeriesPercentage;
+    totalSeriesPercentage += parseFloat(eachSeriesPercentage);
+}
+
+return {
+  percentageBySeriesDays,
+  totalSeriesPercentage,
+};
+}
+
+  // RENDER
+  function present(percentages) {
+  for (const [title, percentage] of Object.entries(percentages.percentageBySeriesDays)) {
```

Comment:
Wow, very nice! Let me know if you have questions about this syntax (it is called object destructuring)

Reply:
I definitely do, I applied this bc Milton suggested in the elastic room. 

It's clear to me that the `of Object.entries(percentages.percentageBySeriesDays)` is accessing one of the returned variables from the previous function. I can "see" that `const [title, percentage] ` is defining how the key and the value inside the object are going to be called in this for loop. How does the for loop know that `[title, percentage]` refers to each key-value pair? 

Besides, I'm still investigating this "built in?" functions that start with Object.something. `Object.entries()`,  `Object.properties()`, and  `Object.values()`. Milton also mentioned something static and dynamic which I still don't see clearly.